### PR TITLE
Change minimum depth to 0 for `ConvertTo-Json`

### PR DESCRIPTION
**Pull request by kvprasoon**
Recreating PR https://github.com/PowerShell/PowerShell/pull/13700

# PR Summary

Change min depth to 0 for ConvertTo-Json

Related to:  https://github.com/PowerShell/PowerShell/issues/13660

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/PowerShell/PowerShell/issues/13660
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

**Comment by kvprasoon** [comment]
Cc: @TravisEz13

**Comment by TravisEz13** [comment]
@SteveL-MSFT  can you code review this?

**Comment by TravisEz13** [comment]
What is the purpose of a 0 depth?

**Comment by kvprasoon** [comment]
> What is the purpose of a 0 depth?

Its based on https://github.com/PowerShell/PowerShell/issues/13660

**Comment by iSazonov** [comment]
@kvprasoon Is there an related issue to add in the description?

**Comment by kvprasoon** [comment]
@iSazonov Yes. Its already there in PRSummary

**Comment by iSazonov** [comment]
> @iSazonov Yes. Its already there in PR Summary

I don't see :-(

**Comment by vexx32** [comment]
Related issue is https://github.com/PowerShell/PowerShell/issues/13660

**Comment by kvprasoon** [comment]
> I don't see :-(

PR CheckList actually, ;-).

**Comment by TravisEz13** [comment]
In your doc issue you describe the behavior of depth 2.  Not depth 0.  I see no need for this PR.

**Comment by kvprasoon** [comment]
Actually I'm not the one who raised the issue. But I support that issue and hence created the PR.
If not value. Then I am okay to close this :-).

**Comment by vexx32** [comment]
This comment shows the need for this: https://github.com/PowerShell/PowerShell/issues/13660#issuecomment-699669594

Basically it forces the json to a "flat" object. There's no reason _not_ to allow this, and although it may be considered edge case for some, I see no reason to force folks to find weird workaround when we can just fix it. 🙂

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1 [hunk 1]
```diff
@@ -40,6 +40,28 @@ Describe 'ConvertTo-Json' -tags "CI" {
         $ps.Dispose()
     }
 
+    It "Should accept minimum depth as 0." {
+        $ComplexObject = [PSCustomObject] @{
+            FirstLevel1  = @{
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1:45]: Can you include a value type in the test?

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1 (hunk 1 above)
Comment (kvprasoon) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1:45]: @TravisEz13  Done

**Review by TravisEz13 — APPROVED** [review]
